### ISSUE TITLE
Support `attachMap` and `attachSeed` independently of `attach`

### ISSUE THREAD

**Issue by maloewe-ona** [wvengen/proguard-maven-plugin#288]
Currently the parameters `attachMap` and `attachSeed` only have an effect if `attach` is set as well.
However, there might be cases where it would be useful to replace the final JAR file (i.e. `attach=false`), but still use `attachMap=true` to attach the mappings to make troubleshooting later easier.

Would it be possible to decouple these options so their are independent of `attach`?

### Workaround
Set `attach=true` and `appendClassifier=false`, and then `attachMap=true` or `attachSeed=true`.

It appears this will then make the plugin use the same output name as the original JAR file, therefore replacing it and not additionally attaching the processed JAR.
But that feels rather hacky and I am not sure if that is officially supported usage of the plugin, since it defeats the purpose of `attach=true`.

**Comment by lasselindqvist** [comment]
So attach is related to the outJar/inJar parameters.
For outJar the docs say: "If attach=true the value ignored and name constructed base on classifier. If empty input jar would be overdriven."

Attaching is not really needed for anything but automatically deploying the produced artifacts to a repository. You can leave attach as false and manually use https://www.mojohaus.org/build-helper-maven-plugin/attach-artifact-mojo.html to attach whatever artifacts you wish.

So also if you want to obfuscate, but not deploy the obfuscated JAR, you could attach the seed and map files with the build-helper-maven-plugin.

In your case where you obfuscate and replace the original JAR and attach and deploy the JAR, seed an map, I would say the configuration you use is correct and fine.

**Comment by lasselindqvist** [comment]
Yes, I think we could just move the attachMap and attachSeed conditions to be fully separate from attach condition. It make make sense to rename attach to be attachJar then for consistency, but that would be a breaking change.

**Comment by lasselindqvist** [comment]
I think that is a good strategy:
1. Deprecate attach.
2. Add attachJar parameter
3. Make both Big Booleans. If attachJar is not-null, use that value. If null, use attach if not-null. If both are null, default to the current default, which is false.
4. In addition to point 3, always log warning if attach is not-null.
5. In a later version, remove attach parameter.
